lib/salus/scanners/base.rb

```diff
@@ -76,11 +79,24 @@ def run!(salus_report:, required:, pass_on_raise:, reraise:)
       salus_report.add_scan_report(@report, required: required)
 
       begin
-        @report.record { run }
+        @report.record do
+          Timeout.timeout(max_lifespan / 1000.to_f) { run }
+        end
 
         if @report.errors.any?
           pass_on_raise ? @report.pass : @report.fail
         end
+      rescue Timeout::Error => e
+        timeout_error_data = {
+          message: "Scanner #{name} timed out during execution",
+          error_class: ScannerTimeoutError,
+          backtrace: e.backtrace.take(5)
+        }
+        @report.error(timeout_error_data)
+        salus_report.error(timeout_error_data)
+
+        # Propagate this error if desired
+        raise ScannerTimeoutError, timeout_error_data.message if reraise
```

Comment:
If you raise an Exception, the salus will fail.
We don't want salus to fail if it times out.  We just want to log the timeout error.  Please also log with `bugsnag_notify`